### PR TITLE
feat: Add DataMatcher search strategy

### PR DESCRIPTION
**Pull request by dpgraham**
* Add data matcher as search strategy
* Added client recording code ahead of https://github.com/appium/appium/issues/12372

### CODE REVIEW

File: app/renderer/lib/client-frameworks/java.js [hunk 1]
```diff
@@ -67,11 +67,16 @@ ${this.indent(code, 4)}
...
       '-android uiautomator': 'AndroidUIAutomator',
+      '-android datamatcher': 'AndroidDataMatcher',
       '-ios predicate string': 'IosNsPredicate',
       '-ios class chain': 'IosClassChain',
     };
     if (!suffixMap[strategy]) {
-      throw new Error(`Strategy ${strategy} can't be code-gened`);
+      if (isArray) {
+        return `List<MobileElement> ${localVar} = (MobileElement) driver.findElementsBy('${strategy}', ${JSON.stringify(locator)});`;
+      } else {
```
Comment (mykola-mokhnach) [inline app/renderer/lib/client-frameworks/java.js]: else is redundant

File: app/renderer/lib/client-frameworks/robot.js [hunk 2]
```diff
@@ -55,7 +55,8 @@ ${this.indent(code, 4)}
       'id': 'id',
       'name': 'name', // TODO: How does Python use name selector
       'class name': 'class_name',
-      '-android uiautomator': 'AndroidUIAutomator',
+      '-android uiautomator': 'unsupported',
```
Comment (mykola-mokhnach) [inline app/renderer/lib/client-frameworks/robot.js]: should this be translated?

Comment (dpgraham) [reply app/renderer/lib/client-frameworks/robot.js]: Yeah the old one was incorrect

Comment (mykola-mokhnach) [reply app/renderer/lib/client-frameworks/robot.js]: then `t('unsupported')`?

**Pull request merged**